webhook_handlers/views/gitlab.py

```diff
@@ -129,112 +155,92 @@ def _handle_merge_request_event(self, repo):
             try:
                 owner = Owner.objects.get(
-                    service=self.service_name, username=owner_username
-                )
-
-                obj, created = Repository.objects.get_or_create(
-                    author=owner,
-                    service_id=self.request.data.get("project_id"),
-                    name=repo_name,
-                    private=self.request.data.get("project_visibility") == "private",
+                    service=self.service_name,
+                    oauth_token__isnull=False,
+                    name=owner_name,
+                    email=owner_email,
                 )
-                message = "Repository created"
+                self._initiate_sync_for_owner(owner)
             except Owner.DoesNotExist:
-                message = "Repository not created - unknown owner"
+                return
+
+    def _handle_system_project_create_hook_event(self):
+        self._try_initiate_sync_for_owner()
+        return Response(data="Sync initiated")
+
+    def _handle_system_hook_event(self, repo, event_name):
+        """
+        GitLab Enterprise instance can send system hooks for changes on user, group, project, etc
+        """
+        message = None
 
-        elif event_name == "project_destroy":
+        if event_name == "project_destroy":
             repo.deleted = True
             repo.activated = False
             repo.active = False
             repo.name = f"{repo.name}-deleted"
             repo.save(update_fields=["deleted", "activated", "active", "name"])
             message = "Repository deleted"
 
-        elif event_name == "project_rename":
-            new_name = self.request.data.get("path_with_namespace").split("/")[-1]
-            repo.name = new_name
-            repo.save(update_fields=["name"])
-            message = "Repository renamed"
-
-        elif event_name == "project_transfer":
-            owner_username, repo_name = self.request.data.get(
-                "path_with_namespace"
-            ).split("/")
-            new_owner = Owner.objects.filter(
-                service=self.service_name, username=owner_username
-            ).first()
-
-            if new_owner:
-                repo.author = new_owner
-                repo.name = repo_name
-                repo.save(update_fields=["author", "name"])
-            message = "Repository transfered"
-
-        elif event_name == "user_create":
-            obj, created = Owner.objects.update_or_create(
-                service=self.service_name,
-                service_id=self.request.data.get("user_id"),
-                username=self.request.data.get("username"),
-                email=self.request.data.get("email"),
-                name=self.request.data.get("name"),
-            )
-            message = "User created"
+        elif event_name in ("project_rename", "project_transfer"):
+            self._try_initiate_sync_for_owner()
```

Comment:
so my current hypothesis is this - and i think we were thinking about it wrong before:
A user can only transfer a repo if they have access to the previous location and the new location (both OwnerOrgs).
So doing a sync on the OwnerUser making the transfer request should give us access to both projects (OwnerOrgs), and the sync process must be able to self-correct project/group structure and organization because that's the sync function we use when the user isn't seeing their repos correctly...
So I think this sync (on the OwnerUser) is able to initiate a sync for both OwnerOrgs :)